x-pack/plugins/triggers_actions_ui/public/common/types.ts

```diff
@@ -5,17 +5,29 @@
  * 2.0.
  */
 
+import { KBN_FIELD_TYPES } from '@kbn/field-types';
+
 export interface Comparator {
   text: string;
   value: string;
   requiredValues: number;
 }
 
+export type ValidNormalizedTypes = `${Exclude<
```

Comment:
First usage like this I've seen of template literal types: https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html :-)

Had to load it up in vscode and hover to see exactly what it was doing, heh.